lib/index.js

```diff
@@ -0,0 +1,225 @@
+    if (_.isString(JSONAttrName)) {
+      Object.assign(result, {
+        [JSONAttrName]: vCardAttrName,
+      });
+    }
+
+    return result;
+  }, {});
+
+  Object.assign(this, {
+    complexJSONAttributes,
+    mappings: {
+      toJSON,
+      toVcard,
+    },
+  });
+}
+
+// Debugging
+VCardParser.debug =
+  process.env.NODE_DEBUG && /\boniyi-locker\b/.test(process.env.NODE_DEBUG);
+
+function debug(args) {
+  if (VCardParser.debug) {
+    console.error('VCardParser %s', util.format(...args));
+  }
+}
+
+/*
+ * @ description: Parses a vCard 2.1 string and creates a JSON-Object with editableField names as properties containing the values from according vCardAttributes
+ * @ param: {string} vCardStr - The vCard string
+ *
+ * @ return {json}: a structured JSON-Object containing the corresponding profile attributes
+ *
+ * @ author: Benjamin Kroeger < benjamin.kroeger@de.ibm.com >
+ */
+VCardParser.prototype.toObject = function toObject(vCardStr, encode) {
+  const self = this;
+  const { mappings: { toJSON }, complexJSONAttributes } = self;
+  const jsonObj = {};
+  const trimmedVCardStr = vCardStr.trim();
+  // split the string on "\n" to receive singular vCardAttributes
+  const validVCardAttributes = [];
+
+  // see if the vCardAttribute starts with one of the configured JSONAttribute-mappings
+  trimmedVCardStr.split('\n').forEach((vCardLine) => {
+    let valid = false;
+    _.forOwn(toJSON, (JSONAttrName, vCardAttrName) => {
+      if (vCardLine.indexOf(vCardAttrName) === 0) {
+        valid = true;
+        if (_.isString(JSONAttrName) && JSONAttrName.length > 0) {
+          validVCardAttributes.push(vCardLine);
+        }
+      }
+    });
+    if (
+      !valid &&
+      _.isString(validVCardAttributes[validVCardAttributes.length - 1])
+    ) {
+      // no mapping was found; either it has not been configured or the value from the previous vCardAttribute contained a "\n". To prevent from data-loss, value is appended to the previous attribute.
+      validVCardAttributes[validVCardAttributes.length - 1] += ` ${vCardLine}`;
+    }
+  });
+
+  // iterate over identified valid vCardAttributes and split them into key / value to create JSON-Object
+  validVCardAttributes.forEach((vCardEntry) => {
+    const trimmedVCardEntry = vCardEntry.trim();
+    if (trimmedVCardEntry.length > 1) {
+      const vCardRex = trimmedVCardEntry.match(keyValuePairRegex);
+      const JSONAttrName = toJSON[vCardRex[1]];
+
+      if (_.isString(JSONAttrName)) {
+        // this vCardAttribute is an extension-attribute from IBM Connections
+        if (JSONAttrName === 'extattr') {
+          try {
+            jsonObj.extattr = jsonObj.extattr || [];
+            const subValues = vCardRex[2].split(';');
+
+            const extAttr = {
+              id: subValues.shift(),
+            };
+
+            subValues.forEach((subValue) => {
+              const [, key, value] = subValue.match(keyValuePairRegex);
```

Comment:
what happens now if the regex doesn't match? I assume, the array destructuring would just throw an error.
How about using `.split(':')` again and destructure it with the spread operator?

```
const [key, ...values] = subValue.split(':');
```

also check if extAttrSubvaluesMapping for key exists
and join values with `values.join(':')`

Reply:
yes, absolutely right. Using destructuring with `.split(':')` is much safer.